File: CodeSignal/Python/adjacentElementsProduct.py
"""
Given an array of integers, find the pair of adjacent elements that has the largest product and return that product.

Example

For inputArray = [3, 6, -2, -5, 7, 3], the output should be
solution(inputArray) = 21.

7 and 3 produce the largest product.
"""
def solution(inputArray):
    largestNum = float("-inf")
    for i in range(0, len(inputArray)-1):
        if inputArray[i] * inputArray[i+1] > largestNum:
            largestNum = inputArray[i] * inputArray[i+1]
    return largestNum
print(solution(inputArray = [3, 6, -2, -5, 7, 3]))

File: CodeSignal/C++/adjacentElementsProduct.cpp
#include <vector>
#include <iostream>
#include <algorithm>
int solution(std::vector <int> inputArray);
int main() {
	std::vector <int> inputArray = {3, 6, -2, -5, 7, 3};
	std::cout << "The solution is " << solution(inputArray); 
}
int solution(std::vector <int> inputArray) {
	int ans = -987654321;
	for (int i = 1; i < inputArray.size(); i++) {
		ans = std::max(ans, inputArray[i] * inputArray[i-1]);
	}
	return ans;
}